pkg/storage/storage.go

```diff
@@ -80,69 +84,121 @@ func (p *storage) GetContainerMetrics(pods ...apitypes.NamespacedName) ([]api.Ti
 
 	timestamps := make([]api.TimeInfo, len(pods))
 	resMetrics := make([][]metrics.ContainerMetrics, len(pods))
-
-	for i, pod := range pods {
-		metricPoint, present := p.pods[pod]
+	for podIdx, pod := range pods {
+		contPoints, present := p.pods[pod]
 		if !present {
 			continue
 		}
 
-		contMetrics := make([]metrics.ContainerMetrics, len(metricPoint.Containers))
-		var earliestTS *time.Time
-		for i, contPoint := range metricPoint.Containers {
-			contMetrics[i] = metrics.ContainerMetrics{
+		contMetrics := make([]metrics.ContainerMetrics, len(contPoints))
+		var earliestTS time.Time
+		var contIdx int
+		for contName, contPoint := range contPoints {
+			prevContPoint := p.prevPods[pod][contName]
```

Comment:
Same comment as with nodes, what behaviour we expect when we are missing data from previous Pod?

Reply:
I updated this so that if we are missing the previous container data, we skip the container completely.